### PR TITLE
No more monoids

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs [hunk 1]
```diff
@@ -64,7 +66,7 @@ data Segment k = Segment { low :: !k, high :: !k }
 point :: k -> Segment k
 point k = Segment k k
 
-data Node k a = Node !(Segment k) !a
+data Node k a = Node !k !a
```
Comment (jianwan2016) [inline src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs]: We need a `Foldable` for Node too?

File: src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs [hunk 2]
```diff
@@ -97,59 +111,75 @@ instance Foldable (SegmentMap k) where
...
        => Segment k
        -> Maybe a
        -> SegmentMap k a
        -> SegmentMap k a
 update (Segment lo hi)   _        m | lo > hi    = m
 update _                 Nothing  m              = m
-update s@(Segment lo hi) (Just x) (SegmentMap t) =
-  SegmentMap $ cappedL lo lt >< Node s x <| cappedR hi rt
+update s@(Segment lo hi) (Just x) (SegmentMap (OrderedMap t)) =
+  SegmentMap $ OrderedMap (cappedL lo lt >< Node (Min lo) (s, x) <| cappedR hi rt) -- Node s x <| cappedR hi rt
```
Comment (Kheldar) [inline src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs]: Lose the comment?

File: src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs [hunk 3]
```diff
@@ -97,59 +111,75 @@ instance Foldable (SegmentMap k) where
...
   where
-    (lt, ys) = FT.split (>= lo) t
-    (_, rt)  = FT.split (> hi) ys
+    (lt, ys) = FT.split (>= Min lo) t
+    (_, rt)  = FT.split (> Min hi) ys
 
-cappedL :: (Enum k, Monoid k, Ord k) => k -> FingerTree k (Node k a) -> FingerTree k (Node k a)
+cappedL :: (Enum k, Ord k, Bounded k)
+  => k
+  -> FingerTree (Min k) (Node (Min k) (Segment k, a))
```
Comment (Kheldar) [inline src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs]: Do these types reduce to `SegmentMap k a`?

**Comment by Kheldar** [comment]
LGTM, but we need to make getting some unit tests working our next priority, because we still have no idea if this actually works.

File: src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs (hunk 3 above)
Comment (newhoggy) [inline src/HaskellWorks/Data/SegmentMap/FingerTree/Strict.hs]: Almost.  `SegmentMap k a` has two additional `newtype` wrappers.

**Pull request merged**